src/Catalog/Registration/RegistrationKey.cs

```diff
@@ -34,7 +34,7 @@ public override bool Equals(object obj)
                 return false;
             }
 
-            return (Id == rhs.Id); 
+            return StringComparer.OrdinalIgnoreCase.Equals(Id, rhs.Id); 
```

Comment:
we need tests for this class that make sure that if 2 objects are equal with Equals, GetHashCode also equals